main.py

```diff
@@ -109,31 +105,36 @@ def execute(self):
                     ethernet.source, ethernet.destination,
                     switch.dpid, interface.port_number)
 
-    @listen_to('kytos/of_core.handshake.completed')
-    def install_lldp_flow(self, event):
-        """Install a flow to send LLDP packets to the controller.
+    @listen_to('kytos/topology.switch.(enabled|disabled)')
+    def handle_lldp_flows(self, event):
+        """Install or remove flows in a switch.
 
-        The proactive flow is installed whenever a switch connects.
+        Install a flow to send LLDP packets to the controller. The proactive
+        flow is installed whenever a switch is enabled. If the switch is
+        disabled the flow is removed.
 
         Args:
             event (:class:`~kytos.core.events.KytosEvent`):
                 Event with new switch information.
 
         """
         try:
-            of_version = event.content['switch'].connection.protocol.version
+            dpid = event.content['dpid']
+            switch = self.controller.get_switch_by_dpid(dpid)
+            of_version = switch.connection.protocol.version
+
         except AttributeError:
             of_version = None
 
-        flow_mod = self._build_lldp_flow_mod(of_version)
-
-        if flow_mod:
-            name = 'kytos/of_lldp.messages.out.ofpt_flow_mod'
-            content = {'destination': event.content['switch'].connection,
-                       'message': flow_mod}
-
-            event_out = KytosEvent(name=name, content=content)
-            self.controller.buffers.msg_out.put(event_out)
+        flow = self._build_lldp_flow(of_version)
+        if flow:
+            destination = switch.id
+            endpoint = f'{settings.FLOW_MANAGER_URL}/flows/{destination}'
+            data = {'flows': [flow]}
+            if event.name == 'kytos/topology.switch.enabled':
+                requests.post(endpoint, json=data)
+            else:
+                requests.delete(endpoint, json=data)
```

Comment:
Is there a specific reason to change this from the event bus to the REST API? If not, I'd rather keep it in the internal event bus.

Reply:
Yes, today the NApp `flow_manager` does not support the installation of flows using  Kytos Events.

Reply:
An issue was created to add this support https://github.com/kytos/flow_manager/issues/104